### ISSUE TITLE
 Adding New Outlines via the "Add Dictionary" Command Doesn't Work

### ISSUE THREAD

**Issue by gdwaynewarner** [openstenoproject/plover#1526]
**Describe the bug**
I hit the "Make Dictionary" stroke (PHA*-EBGD), write in what needs to be added, but the "OK" button is grayed out.

**To Reproduce**
Stroke the "Add Dictionary" stroke, type in the necessary information, reach for the Okay button, and try to press it.

**Expected behavior**
Everything looks good, but I can't press the Okay button, so I have to add everything manually.

**Screenshots**
![Screen Shot 2022-05-20 at 1 34 59 PM](https://user-images.githubusercontent.com/17680111/169609267-1ce91a13-fed5-4f0e-950c-59c54678e36f.png)

![Screen Shot 2022-05-20 at 1 42 44 PM](https://user-images.githubusercontent.com/17680111/169609058-55582381-c387-4016-8877-6e267cb67e59.png)

**Operating system**
Mac OS 10.13.6
 - Plover Version [e.g. 3.x, 4.x, weekly]
 - 4.0.0.dev11+21.g486a032 
**Hardware**

If applicable, let us know what machine you're experiencing the issue with and how it's connected to Plover, as well as a screenshot of the machine configuration inside Plover.

Using my original ProCAT Flash.

Note that adding new outlines worked just fine on my laptop which was running the same build, but that's in the shop right now.

**Comment by aerickt** [comment]
I believe this sometimes happens because there is an invisible character in the strokes field which is invalid steno. I've found a workaround to this would be to backspace everything in the strokes field, and then write it again.

**Comment by benoit-pierre** [comment]
I can't even copy-paste an invalid stroke in the steno field, so I don't know how you'd end up with extraneous spaces.

The only way to get a grayed out "Ok" for me is for example an initial `-` (as by itself `-` is not a valid stroke, but you need to be allowed to enter it so you can get to a valid stroke like `-F`). Or a trailing `-`.
 
@gdwaynewarner: how did you enter the stroke? Manually?

**Comment by gdwaynewarner** [comment]
I try to avoid doing anything manually! I just stroke what I wanted to add to the dictionary in a text file, pressed the command to add that stroke to the dictionary, wrote the desired translation with the writer, and tried to press the OK button, with no luck.

Just to be sure, I tried another outline I didn't really need, ensured there were no spaces where there were not supposed to be any, and sure enough, a grayed out "Okay" button.

![Screen Shot 2022-05-20 at 6 49 19 PM](https://user-images.githubusercontent.com/17680111/169629915-ca6e93df-8af6-436a-822c-c30d15915ab9.png)

**Comment by benoit-pierre** [comment]
So that the issue, using `ST*FBG {PLOVER:ADD_TRANSLATION}`, instead of the reverse.

**Comment by gdwaynewarner** [comment]
So with the new build (4.0.0.dev11+23.g991d334), I just tried to add the name "Cristina," using the normal outline for "Christina," which for me is KREUS TAOEPB AU, which I wanted to follow with Phoenix Theory's famous R-R stroke to get the variant I wanted.

When I attempted to add that to the dictionary with the "Add Dictionary" stroke, the window opened just fine, but there were a few things missing:

![cristina_1](https://user-images.githubusercontent.com/17680111/177212696-0f2edaee-3e7c-4d0e-839e-1b86e2e96d04.png)

Since that didn't work, I decided to just add the outline manually -- as in typing it in directly. Unfortunately, that didn't work either:

![cristina_2](https://user-images.githubusercontent.com/17680111/177213187-9118c60e-d9b3-47fd-89e0-e4f82f421e7d.png)


That is to say, I could not write directly into the field, beyond what's there.

You should also note that the "Okay" button is grayed out.

**Comment by benoit-pierre** [comment]
> So with the new build (4.0.0.dev11+23.g991d334), I just tried to add the name "Cristina," using the normal outline for "Christina," which for me is KREUS TAOEPB AU, which I wanted to follow with Phoenix Theory's famous R-R stroke to get the variant I wanted.
> 
> When I attempted to add that to the dictionary with the "Add Dictionary" stroke, the window opened just fine, but there were a few things missing:

What's missing? The dialog will pre-populate either:
- the strokes field if the last translation was an untranslate
- or the translation field (not both)

> Since that didn't work, I decided to just add the outline manually -- as in typing it in directly. Unfortunately, that didn't work either:
>
> That is to say, I could not write directly into the field, beyond what's there.

Where exactly are you trying to write, and what? The entry will not allow invalid steno, except for some transient states (like a trailing `/`, which is necessary to be able to enter multi strokes) that will prevent adding the mapping (grayed out "OK").

As a result, trying to write before `R-R` is going to be problematic; as you'd need a way to insert an initial `/` first, then jump at the start again, before writing the beginning of the outline (`KREUS TAOEPB AU`).

Much easier to just stroke the whole outline from start when the dialog shows up, (the first stroke will erase the already selected `R-R`).

> You should also note that the "Okay" button is grayed out.

Yes, since this issue is still open, the bug is still present in the latest build.

**Comment by gdwaynewarner** [comment]
I was writing into a text file, which is that bit down below the "Add Translation" window in the screenshots in the bug report. Note that I had to turn off all of my dictionaries except for the "empty"  dictionary I have for those times when I need to write something that should be in steno to get the raw steno to appear ... like for this bug report.

So that KR-R outline is all that remains of the "Christina R-R" outline that I was trying to add so it would show up as "Cristina."

Just now, I tried adding just raw steno (KREUS TAOEPB AU) to get rid of the English ("Christina") that appeared, but had the same result:

![cristina_3](https://user-images.githubusercontent.com/17680111/177234638-25178171-dcef-4968-bb19-107526cfc1be.png)

That is, all I got that time was the "R-R" stroke, despite trying to add the outline simply by adding raw steno.

> Yes, since this issue is still open, the bug is still present in the latest build.

Yeah, I forgot about that bug ... pretty difficult to do anything without a working "Okay" button!
